Makefile

```diff
@@ -156,8 +159,19 @@ test-integration: manifests generate envtest ginkgo mpi-operator-crd ray-operato
 
 CREATE_KIND_CLUSTER ?= true
 .PHONY: test-e2e
-test-e2e: kustomize manifests generate envtest ginkgo
-	E2E_KIND_VERSION=$(E2E_KIND_VERSION) KIND_CLUSTER_NAME=$(KIND_CLUSTER_NAME) CREATE_KIND_CLUSTER=$(CREATE_KIND_CLUSTER) ARTIFACTS=$(ARTIFACTS) IMAGE_TAG=$(IMAGE_TAG) ./hack/e2e-test.sh
+test-e2e: kustomize manifests generate envtest ginkgo run-test-e2e-$(E2E_KIND_VERSION:kindest/node:v%=%)
+
+E2E_TARGETS := $(addprefix run-test-e2e-,${E2E_K8S_VERSIONS})
+.PHONY: test-e2e-all
+test-e2e-all: kustomize manifests generate envtest ginkgo cleanup-e2e $(E2E_TARGETS)
+
+.PHONY: cleanup-e2e
+cleanup-e2e:
+	rm -f $(E2E_TARGETS) || echo "ignoring remove errors"
```

Comment:
What is this remove?

Reply:
I was unable to make `.PHONY` work properly with the wildcard targets.

Reply:
Ah, I see.

Reply:
which is why a `foreach` + `($MAKE)` might work better than using targets :)

Reply:
the foreach + ($MAKE) is not helping with the `.PHONY` issue, the target will be the target will be the same and the potential existence of a file with that name will skip the target execution.

Anyway, I found a way around the need of  `cleanup-e2e`.